### DISCUSSION TITLE
Using dynamic error codes?

### DISCUSSION THREAD

**Discussion by dickermoshe** [invertase/dart_custom_lint#277] [General]
The docs for `LintRule` state that `[LintRule]s should avoid emitting lints that use a code different that [code].`
https://github.com/invertase/dart_custom_lint/blob/90365e2c47526587603f9132cf89eb7d352ee057/packages/custom_lint_core/lib/src/lint_rule.dart#L64-L67

Q1: Why?
Q2: What about if the code name stays the same but the `problemMessage` changes

Use Case: We would like to use our code generator (drift_dev) with the custom_lint to find errors. The errors emitted can be of many types.

**Comment by rrousselGit** [comment]
> Q1: Why?

To disable lints in `analysis_options.yaml` and when writing `// ignore_for_file: code`.

> Q2: What about if the code name stays the same but the problemMessage changes

That's fine. But LintCode already supports dynamic messages.

You can do `problemMessage: 'Hello {0}'`

then used as `errorReporter.atNode(node, arguments: ['msg'])` // The message will be `Hello msg`

**Comment by dickermoshe** [comment]
@rrousselGit Have you given any thought about using custom_lint on the to build the code live instead of using `build_runner`?

Would Seem like macros!

**Reply by rrousselGit** [reply]
I have no clue what you're referring to. Custom_lint doesn't even use build_runner :)

**Reply by dickermoshe** [reply]
## Background

The code generator for drift is pretty flexible.
Our main implementation of the code generator works with `build_runner`, which uses all the great tools `build_runner` has to offer for parsing and navigating dart code.

However we built a second code generator which works with just a simple `AnalysisSession`. 

## But why?

Well, when we started writing a `custom_lint` plugin, we realized, "Hey, our code generator already parses the source code and spits out errors, why are we doing writing this twice?!

So we built a version of our code generator which can work with what `custom_lint` gives us.

Now we can extract all of these errors and put them we they need to be
<img width="512" alt="image" src="https://github.com/user-attachments/assets/2ef6ecb6-dc31-49fa-9447-8bf9f27070ff">

## Results

Now errors which would normally only be emitted when running `dart run build_runner build`, are show in the editor!

<img width="607" alt="image" src="https://github.com/user-attachments/assets/25567dee-c248-4215-9cdb-e1c6ea03e04d">


## Idea

Instead of just running the code generator under the hood to find errors, why not actually write the code too!
